graph/run-dynamic.go

```diff
@@ -280,17 +281,22 @@ func ConvergeDynamicThread[EP EPP[E], V VPI[V], E EPI[E], M MVI[M], N any, A Alg
 			gt.Status = APPLY_MSG
 			checkTerm := (strucClosed && remitClosed) || (epoch && remitCount == 0 && topCount == 0)
 			completed, algCount = ProcessMessages[V, E, M, N](alg, g, gt, checkTerm)
-			if completed && epoch {
-				//log.Debug().Msg("T[" + utils.F("%02d", tidx) + "] completed epoch")
-				gt.Status = DONE
-				gt.Response <- ACK
-
-				resp := <-gt.Command // BLOCK and wait for resume
-				if resp != RESUME {
-					log.Panic().Msg("Expected to resume after blocked")
+			if completed { // no backoff when completed is true
+				if checkSuperStep {
+					completed = AwaitSuperStepConvergence[V, E, M, N](alg, g, tidx)
+				}
+				if epoch {
```

Comment:
Hmm, maybe I misunderstand something. But my assumption for an Algorithmic Super-step is that, if it creates new messages -- the algorithm is saying "please continue processing, I am not actually complete".

In this case, I think we should get out of this branch as we are actually not ready to report being done to the "epoch"

Reply:
Oops. There was a `if !completed {continue}` right after `AwaitSuperStepConvergence` but somehow it was removed.